tests/unittests/tests-netutils/mock_dns.c

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (C) 2021 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Mock implementation of sock_dns
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ */
+#include <string.h>
+
+#include "net/af.h"
+#include "net/ipv6/addr.h"
+#include "net/sock/dns.h"
+
+int sock_dns_query(const char *domain_name, void *addr_out, int family)
```

Comment:
This could easily lead to problems once some other test suite in this application for one reason or another decides to include `sock_dns`. IMHO we have three options here:

1. Don't test the DNS functionality
2. Move the `tests-netutils` test suite to its own application
3. Keep the non-DNS tests here and test the DNS functionality in its own application

the latter would have the added bonus that all compile-time configs of the module would be tested.